File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/state/NoonState.java
package jv.com.tencent.cain.design_pattern.state;

/**
 * @author cainjiang
 * @date 2020/3/27
 */
public class NoonState extends State {

    @Override
    public void doWork(Work work) {
        if (work.getHour() < 13) {
            System.out.println("当前时间：" + work.getHour() + "属于午休时间");
        } else {
            work.setState(new AfterNoonState());
            work.doWork();
        }
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/state/NoonState.kt
package kt.com.tencent.cain.desigin_pattern.state

/**
 * @author : jiangyu
 * @date   : 2020/3/28
 * @desc   : xxx
 */
class NoonState : State() {
    override fun doWork(work: Work) {
        if (work.curTimeHour <= 13) {
            println("当前时间：${work.curTimeHour}点，属于午休时间")
        } else {
            work.state = AfterNoonState()
            work.doWork()
        }
    }
}